overlord/ifacestate/helpers.go

```diff
@@ -357,22 +357,27 @@ func (m *InterfaceManager) removeConnections(snapName string) error {
 }
 
 func (m *InterfaceManager) setupSecurityByBackend(task *state.Task, snaps []*snap.Info, opts []interfaces.ConfinementOptions, tm timings.Measurer) error {
+	if len(snaps) != len(opts) {
+		return fmt.Errorf("internal error: setupSecurityByBackend received an unexpected number of snaps (expected: %d, got %d)", len(opts), len(snaps))
+	}
+	confOpts := make(map[string]interfaces.ConfinementOptions, len(snaps))
+	for i, snapInfo := range snaps {
+		confOpts[snapInfo.InstanceName()] = opts[i]
+	}
+
 	st := task.State()
+	st.Unlock()
+	defer st.Lock()
```

Comment:
The lock ends up being in place for a pretty long time in one go, as opposed to before with several smaller locks. I can't see a way around it and am not sure it is a problem, but it might be worth verifying it is not a problem.

Reply:
This is actually not the case ;), but I see how this can be confusing. The function is called with lock in place, and we  unlock the state before calling backend's setup. The state needs to be locked back before returning from the function.

Reply:
Right, I understand that but typed it wrong (erf). I meant to say 'This *un*lock ends up being in place for a pretty long time in one go, as opposed to before with several smaller *un*locks.'

My point was that a) there is more code in each call to SetupMany() than to the previous Setup() in the loop iteration and b) now all iterations of the loop must be gone through before Unlock() whereas before, it was an Unlock()/Lock() on each iteration. I'm not saying this is a problem, just calling it out since I don't know what the side-effects of being unlocked for so long are, if any.

Reply:
Ok, I see. I was thinking if the unlocking/locking there was deliberate or not. Semantically it looks irrelevant to me and if it had any sideffects it would be a bit fishy tbh. Either way it's impossible to test / prove.